### PR TITLE
Easily usable, efficient utility for converting integers to string_view.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
My pull requests do seem to be useful for exercising CI, if nothing else.

File: lib/ts/IntStr.h [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  std::size_t
+  size() const
+  {
+    return _size;
+  }
+
+  ts::string_view
+  sV() const
```
Comment (dragon512) [inline lib/ts/IntStr.h]: I would suggest a better name would be to use as_xyz() for objects with conversions functions to a different object.
so something like:
 as_string_view()
 as_sv()
 asStringView()

would be a better convention to use on all our objects

**Comment by dragon512** [comment]
👍 on having a unit test!!

File: lib/ts/IntStr.h (hunk 1 above)
Comment (ywkaras) [reply lib/ts/IntStr.h]: I think short abbreviations have their place, but I already lost that argument.  If it's going to be long anyway, we could just have the conversion operator, and write ts::string_view(intStr) when we want it to be explicit.  Although @zwoop may insist it be static_cast<ts::string_view>(intStr) .

Comment (dragon512) [reply lib/ts/IntStr.h]: I am just looking for clarity in the functions. I am for names that say intent. asType() is a common pattern that makes it clear what the intent is in the function, which makes it easy to understand when reading. sV() is unclear. Explicit conversion via static_cast or a clearly named function, I am ok with both I would rather have the code read cleaning. I think this I small and a good improvement to the code.

**Comment by dragon512** [comment]
This might be useful... I am not sure, old code I had.

you would want a replacement for typename uint_type<sizeof(_IntType)*8>::least nval;
numerical traits should have an equal for it..

Again just some code that might be your stuff more reusable


template<typename _IntType,typename _CharType>
	void xtostr(const _IntType val, _CharType * const buf,const unsigned radix)
	{


			_CharType *ptr;                // pointer to string
			_CharType *start;         // pointer to first digit
			_CharType temp;              // temp char
			uint8_t digval;        // value of digit ... has to be positive
			typename uint_type<sizeof(_IntType)*8>::least nval;

			ptr = buf;

			if ( val<0 )
			{
				*ptr++ = '-';          // add negative
				// 2 comp of neg can be one larger than the positive can hold
				nval = -val; // make positive
			}
			else{
				nval = val;
			}

			start = ptr;           // save pointer to first digit

			do {
				digval =  static_cast<uint8_t>(nval % radix); // get the digit value
				nval /= radix;       // get next digit

				// convert to ascii and store
				if (digval > 9)
					*ptr++ = digval - 10 + 'a';  // a letter 0-9
				else
					*ptr++ = digval + '0';       // a digit A-??
			} while (nval > 0);

			// We now have the digit of the number in the buffer, but in reverse
			//  order.  Thus we reverse them now.

			*ptr-- = '\0';            // terminate string; p points to last digit

			do {
				temp = *ptr;		// might try std::swap(...) later
				*ptr = *start;
				*start = temp;   // swap *p and *firstdig
				--ptr;
				++start;         // advance to next two digits
			} while (start < ptr); // repeat until halfway
		}

	template<typename _IntType,typename _CharType>
	void uxtostr(const _IntType val, _CharType * const buf,const unsigned radix)
	{
			_CharType *ptr;                // pointer to string
			_CharType *start;         // pointer to first digit
			_CharType temp;              // temp char
			uint8_t digval;        // value of digit ... has to be positive
			_IntType nval= val;


			start = ptr = buf;

			do {
				digval =  static_cast<types::uint8_t>(nval % radix); // get the digit value
				nval /= radix;       // get next digit

				// convert to ascii and store
				if (digval > 9)
					*ptr++ = digval - 10 + 'a';  // a letter 0-9
				else
					*ptr++ = digval + '0';       // a digit A-??
			} while (nval > 0);

			// We now have the digit of the number in the buffer, but in reverse
			//  order.  Thus we reverse them now.

			*ptr-- = '\0';            // terminate string; p points to last digit

			do {
				temp = *ptr;		// might try std::swap(...) later
				*ptr = *start;
				*start = temp;   // swap *p and *firstdig
				--ptr;
				++start;         // advance to next two digits
			} while (start < ptr); // repeat until halfway
		}
`

**Comment by ywkaras** [comment]
@dragon512 Not sure about the use of literals like '0', 'a' when your character type is other than 'char'.  I don't see what it buys to write the digits from least to most significance and then reverse the order.  Use of templates may incur the wrath of @zwoop The Beserker.

**Comment by ywkaras** [comment]
Closing to remove clutter from open PR list.

**Pull request closed without merging**